### PR TITLE
chore: generate docs versions var

### PR DESCRIPTION
**Pull request by rollandf**
Generate documentation var files to be used
in network-operator-docs repo.

### CODE REVIEW

File: hack/templates/docs-versions-var/vars.template [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+.. |multus-version| replace:: {{ .Multus.Version }}
+.. |ipoib-cni-version| replace:: {{ .Ipoib.Version }}
+.. |whereabouts-version| replace:: {{ .IpamPlugin.Version }}
+.. |nvidia-ipam-version| replace:: {{ .NvIPAM.Version }}
+.. |nic-feature-discovery-version| replace:: {{ .NicFeatureDiscovery.Version }}
+.. |sriovnetop-image-tag| replace:: {{ .SriovNetworkOperator.Version }}
+.. |sriovnetop-sriov-cni-image-tag| replace:: {{ .SriovCni.Version }}
+.. |sriovnetop-ib-sriov-cni-image-tag| replace:: {{ .SriovIbCni.Version }}
+.. |sriovnetop-sriov-device-plugin-image-tag| replace:: {{ .SriovDevicePlugin.Version }}
+.. |node-feature-discovery-version| replace:: v0.13.2
```
Comment (rollandf) [inline hack/templates/docs-versions-var/vars.template:19]: Not in `release.yaml`

File: hack/release.go [hunk 2]
```diff
@@ -136,6 +138,10 @@ func main() {
 	if *retrieveSha {
 		resolveImagesSha(&release)
 	}
+	parts := strings.Split(release.NetworkOperator.Version, ".")
+	release.OcpDefaulChannel = fmt.Sprintf("%s.%s", parts[0], parts[1])
```
Comment (e0ne) [inline hack/release.go:142]: Do we want to generate OCP channels list (e.g. `stable,v24.4.1`) too?

File: Makefile [hunk 3]
```diff
@@ -413,6 +413,11 @@ release-build:
 	cd hack && $(GO) run release.go --templateDir ./templates/values/ --outputDir ../deployment/network-operator/
 	cd hack && $(GO) run release.go --templateDir ./templates/config/manager --outputDir ../config/manager/
 
+.PHONY: generate-docs-versions-var
+generate-docs-versions-var:
+	cd hack && $(GO) run release.go --templateDir ./templates/docs-versions-var/ --outputDir .
+	mv hack/vars.yaml hack/vars.rst
```
Comment (e0ne) [inline Makefile:419]: do we want to add `hack/vars.rst` to `.gitignore`?

File: hack/release.go (hunk 2 above)
Comment (rollandf) [reply hack/release.go:143]: We use this one for OpenShift installation via CLI where we need a single channel, not a list:
image

File: .gitignore [hunk 4]
```diff
@@ -30,3 +30,5 @@ testbin
 test
 build
 hack/manifests
+
+hack/vars.rst
```
Comment (adrianchiris) [inline .gitignore:34]: nit: newline

File: hack/release.yaml [hunk 5]
```diff
@@ -77,3 +78,5 @@ rdmaCni:
   image: rdma-cni
   repository: ghcr.io/k8snetworkplumbingwg
   version: v1.2.0
+Nfd:
+  version: v0.13.2
```
Comment (adrianchiris) [inline hack/release.yaml:82]: nit: its v0.15.6

File: hack/release.go (hunk 2 above)
Comment (adrianchiris) [reply hack/release.go:143]: cant we always specify default here ? then install specific operator version ?

File: hack/release.yaml (hunk 5 above)
Comment (rollandf) [reply hack/release.yaml]: Done

File: .gitignore (hunk 4 above)
Comment (rollandf) [reply .gitignore]: Done

**Comment by adrianchiris** [comment]
at some point, i think we would need to carve out docs related scripts/logic and put them in doc repo if we see we get too much of those, relying on release.yaml file from this repo.

**Pull request merged**